elfi/methods/bo/utils.py

```diff
@@ -57,7 +57,10 @@ def minimize(fun, grad, bounds, prior=None, n_start_points=10, maxiter=1000, ran
 
     # Run optimization from each initialization point
     for i in range(n_start_points):
-        result = fmin_l_bfgs_b(fun, start_points[i, :], fprime=grad, bounds=bounds, maxiter=maxiter)
+        if grad is not None:
+            result = fmin_l_bfgs_b(fun, start_points[i, :], fprime=grad, bounds=bounds, maxiter=maxiter)
+        else:
+            result = fmin_l_bfgs_b(fun, start_points[i, :], bounds=bounds, maxiter=maxiter)
```

Comment:
In this case either `fun` should return both the function value and its gradient, or you have to set `approx_grad=True`.

https://docs.scipy.org/doc/scipy-0.19.0/reference/generated/scipy.optimize.fmin_l_bfgs_b.html